modules/kubernetes-addons/crossplane/main.tf

```diff
@@ -43,134 +45,81 @@ resource "time_sleep" "wait_30_seconds" {
-
-  depends_on = [kubectl_manifest.jet_aws_controller_config]
-}
-
-module "jet_aws_provider_irsa" {
-  count = var.jet_aws_provider.enable == true ? 1 : 0
-
-  source                            = "../../../modules/irsa"
-  create_kubernetes_namespace       = false
-  create_kubernetes_service_account = false
-  kubernetes_namespace              = local.namespace
-  kubernetes_service_account        = "${local.jet_aws_provider_sa}-*"
-  irsa_iam_policies                 = concat([aws_iam_policy.jet_aws_provider[0].arn], var.jet_aws_provider.additional_irsa_policies)
-  irsa_iam_role_path                = var.addon_context.irsa_iam_role_path
-  irsa_iam_permissions_boundary     = var.addon_context.irsa_iam_permissions_boundary
-  eks_cluster_id                    = var.addon_context.eks_cluster_id
-  eks_oidc_provider_arn             = var.addon_context.eks_oidc_provider_arn
-  depends_on                        = [kubectl_manifest.jet_aws_provider]
-}
-
-resource "aws_iam_policy" "jet_aws_provider" {
-  count       = var.jet_aws_provider.enable == true ? 1 : 0
-  description = "Crossplane Jet AWS Provider IAM policy"
-  name        = "${var.addon_context.eks_cluster_id}-${local.jet_aws_provider_sa}-irsa"
-  policy      = data.aws_iam_policy_document.s3_policy.json
-  tags        = var.addon_context.tags
-}
-
-resource "kubectl_manifest" "jet_aws_provider_config" {
-  count     = var.jet_aws_provider.enable == true ? 1 : 0
-  yaml_body = templatefile("${path.module}/aws-provider/jet-aws-provider-config.yaml", {})
-
-  depends_on = [kubectl_manifest.jet_aws_provider]
-}
-
 resource "kubernetes_service_account_v1" "kubernetes_controller" {
   metadata {
-    name      = local.kubernetes_provider_sa
+    name      = local.kubernetes_provider.service_account
     namespace = local.namespace
   }
 
   depends_on = [module.helm_addon]
 }
 
 resource "kubectl_manifest" "kubernetes_controller_clusterolebinding" {
-  count = var.kubernetes_provider.enable == true ? 1 : 0
+  count = local.kubernetes_provider.enable == true ? 1 : 0
   yaml_body = templatefile("${path.module}/kubernetes-provider/kubernetes-controller-clusterrolebinding.yaml", {
-    kubernetes-serviceaccount-name = local.kubernetes_provider_sa
     namespace                      = local.namespace
+    cluster-role                   = local.kubernetes_provider.cluster_role
+    kubernetes-serviceaccount-name = local.kubernetes_provider.service_account
   })
   wait = true
 
   depends_on = [module.helm_addon]
 }
 
 resource "kubectl_manifest" "kubernetes_controller_config" {
-  count = var.kubernetes_provider.enable == true ? 1 : 0
+  count = local.kubernetes_provider.enable == true ? 1 : 0
   yaml_body = templatefile("${path.module}/kubernetes-provider/kubernetes-controller-config.yaml", {
-    kubernetes-serviceaccount-name = local.kubernetes_provider_sa
-    namespace                      = local.namespace
+    kubernetes-serviceaccount-name = local.kubernetes_provider.service_account
```

Comment:
```suggestion
    kubernetes-serviceaccount-name = kubernetes_service_account_v1.kubernetes_controller[0].metadata[0].name
```

Reply:
oh wow genius! 
Does this also creates a implicit dependency?